### PR TITLE
doc: move cloud deployment instruction to docs -v2

### PR DESCRIPTION
**Pull request by annastuchlik**
This is V2 of https://github.com/scylladb/scylladb/pull/14108

This commit moves the installation instruction for the cloud from the [website ](https://www.scylladb.com/download/)to the docs.

The scope:

* Added new files with instructions for AWS, GCP, and Azure.
* Added the new files to the index.
* Updating the "Install ScyllaDB" page to create the "Cloud Deployment" section.
* Adding new bookmarks in other files to create stable links, for example, ".. _networking-ports:"
* Moving common files to the new "installation-common" directory. This step is required to exclude the open source-only files in the Enterprise repository.

In addition:
- The Configuration Reference file was moved out of the installation section (it's not about installation at all)
- The links to creating a cluster were removed from the installation page (as not related).

Related: https://github.com/scylladb/scylla-docs/issues/4091

### CODE REVIEW

**Comment by annastuchlik** [comment]
@tzach I've opened a V2 PR to make sure it merges cleanly. The only change compared to the previous iteration is that I've applied your feedback.
Could you review this PR so that it can be merged?

**Comment by annastuchlik** [comment]
@tzach Would it be possible for you to review this PR soon? It must be merged before we move on with migrating the section for Linux packages.

File: docs/getting-started/install-scylla/index.rst [hunk 1]
```diff
@@ -5,33 +5,36 @@ Install ScyllaDB
...
+.. panel-box::
+  :title: On-premise Deployment
+  :id: "getting-started"
+  :class: my-panel
 
+  * Download and install ScyllaDB Server, Drivers and Tools in `ScyllaDB Download Center <https://www.scylladb.com/download/#core>`_
+  * :doc:`ScyllaDB Web Installer for Linux </getting-started/installation-common/scylla-web-installer>`
+  * :doc:`ScyllaDB Unified Installer (relocatable executable) </getting-started/installation-common/unified-installer>`
+  * :doc:`Air-gapped Server Installation </getting-started/installation-common/air-gapped-install>`
+  * :doc:`What is in each RPM </getting-started/installation-common/rpm-info/>`
```
[the remark this answers is not in the document]
Comment (annastuchlik) [reply docs/getting-started/install-scylla/index.rst:39]: I'll be happy to remove it. I'll push a separate commit with the change, but let's do it as part of this PR.

File: docs/getting-started/install-scylla/run-on-aws.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
+==========================
+Launch ScyllaDB on AWS
```
[the remark this answers is not in the document]
Comment (annastuchlik) [reply docs/getting-started/install-scylla/run-on-aws.rst:2]: I don't think we need to be consistent at all times when it comes to page titles vs. file names, but it makes sense in this case. I'll rename the pages to "launch".

**Comment by annastuchlik** [comment]
@tzach The "merge" commits resolve conflicts related to redirections, and nothing else.
The installation part is still ready for review unchanged. Could you have the final look?

image

**Comment by tzach** [comment]
LGTM

**Comment by annastuchlik** [comment]
@scylladb/scylla-maint Please merge.

**Comment by kbr-scylla** [comment]
For the future: we want to avoid merge commits in scylladb.git PRs to keep our git history simple - with a single main "trunk" with linear branches that at some point merge back to the trunk. Merge commits introduce a lot of noise and it's hard to follow what happens.

If there are conflicts, you shoud **rebase** your PR on the current `master` branch.

**Comment by kbr-scylla** [comment]
> a single main "trunk" with linear branches that at some point merge back to the trunk

Illustration (taken from https://trunkbaseddevelopment.com/):
image

**Comment by annastuchlik** [comment]
@kbr-scylla Thanks for pointing that out!
The conflicts were easy to resolve, so I used the "resolve in GitHub" functionality, which was suggested by GitHub. I didn't know (or have forgotten) that using that feature is a bad practice. I won't use it anymore.
